### PR TITLE
Upgrade to Lean (and mathlib) 4.7.0

### PR DESCRIPTION
**Pull request by monsterkrampe**

### CODE REVIEW

File: FormalSystems/Preliminaries/Language.lean [hunk 1]
```diff
@@ -227,7 +234,13 @@ theorem Sigma.kleene_contains_all : ∀(w : Word α), w ∈ (Σ)∗
...
   | x::xs => by
     have ⟨n, hn⟩ := Sigma.kleene_contains_all xs
-    exists n + 1; exists [x]; exists xs
+    exists (n + 1)
+    rw [pow_succ] 
+    conv => right; right; rw [← pow_one (Σ)]
+    rw [npow_mul_comm]
+    simp
+    exists [x]
+    exists xs
```
Comment (monsterkrampe) [inline FormalSystems/Preliminaries/Language.lean]: This part took me some time. There might be an easier way to do this (e.g. by using a theorem from mathlib that I'm not aware of).
I'm also not sure what exact change caused this to break.

Comment (matzemathics) [reply FormalSystems/Preliminaries/Language.lean]: Here is a small simplification. I guess the problem here is, that the simplification performed by `exists` doesn't work anymore (which is likely caused by the changes to the simp tactic).

```suggestion
    exists (1 + n)
    simp [pow_add]
    exists [x]
    exists xs
```

**Review by matzemathics — APPROVED** [review]
Left a comment at the kleene theorem. Looks good otherwise

File: FormalSystems/Preliminaries/Language.lean (hunk 1 above)
Comment (monsterkrampe) [reply FormalSystems/Preliminaries/Language.lean]: Ah, I tried to use `(1+n)` but gave up since the rest did not work immediately. Thanks!

**Pull request merged**